sway-lsp/src/capabilities/code_lens.rs

```diff
@@ -1,15 +1,13 @@
-use std::{path::PathBuf, sync::Arc};
-
+use crate::server_state::RunnableMap;
 use lsp_types::{CodeLens, Url};
+use std::path::PathBuf;
 
-use crate::core::session::Session;
-
-pub fn code_lens(session: &Arc<Session>, url: &Url) -> Vec<CodeLens> {
+pub fn code_lens(runnables: &RunnableMap, url: &Url) -> Vec<CodeLens> {
     let _p = tracing::trace_span!("code_lens").entered();
     let url_path = PathBuf::from(url.path());
```

Comment:
nit: not part of changed code - but realised this may have issues with paths on windows

`PathBuf::from(url.path())` doesn't properly decode URLs, causing path mismatches that break code-lens on some platforms (thinking windows only for now).

Can use `to_file_path` to handle this potentially
```suggestion
    let url_path = match url.to_file_path() {
        Ok(p) => p,
        Err(_) => return Vec::new(), // invalid URL for local file
    };
```
doesnt need to be done in this PR (since unrelated) - could create gh issue for it.

Reply:
Ok, might leave this for now since we dont support windows. Seems like we'll need to do a big sweep for this sort of thing if and when we ever decide to support it.